pkg/descheduler/strategies/node_affinity.go

```diff
@@ -80,8 +85,7 @@ func RemovePodsViolatingNodeAffinity(ctx context.Context, client clientset.Inter
 					node,
 					podutil.WithFilter(func(pod *v1.Pod) bool {
 						return evictable.IsEvictable(pod) &&
-							!nodeutil.PodFitsCurrentNode(pod, node) &&
-							nodeutil.PodFitsAnyNode(pod, nodes)
```

Comment:
`nodeutil.PodFitsAnyNode` needs to stay here as well

Reply:
This is the only place `PodFitsAnyNode()` was called prior to my feature. Originally I had rewrote this function as `PodFitsAnyOtherNode()` and added the ability to check for taints.

I've now added back in `PodFitsAnyNode()` alongside `PodFitsAnyOtherNode()` to restore the exact behavior of `node_affinity.go`. These functions are very similar so maybe eventually we should drop `PodFitsAnyNode()`.